.github/workflows/firebase-hosting-merge-dev.yml

```diff
@@ -1,38 +1,47 @@
-                  VITE_FIREBASE_MESSAGING_SENDER_ID: ${{ secrets.FIREBASE_MESSAGIN_SENDER_ID }}
-                  VITE_FIREBASE_APP_ID: ${{ secrets.FIREBASE_APP_ID }}
-                  VITE_FIREBASE_MEASUREMENT_ID: ${{ secrets.FIREBASE_MEASUREMENT_ID }}
-                  VITE_GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
-                  VITE_APP_ENV: preview
-                  VITE_APP_CHECK_KEY: ${{ secrets.FIREBASE_APP_CHECK_KEY }}
-                  VITE_APPLICATIONS_COLLECTION: ${{ secrets.APPLICATIONS_COLLECTION }}
-                  VITE_TICKETS_COLLECTION: ${{ secrets.TICKETS_COLLECTION }}
-                  VITE_RSVP_COLLECTION: ${{ secrets.RSVPS_COLLECTION }}
-            - uses: FirebaseExtended/action-hosting-deploy@v0
-              with:
-                  repoToken: "${{ secrets.GITHUB_TOKEN }}"
-                  firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_HAWKHACKS_DASHBOARD }}"
-                  projectId: "${{ secrets.FIREBASE_PROJECT_ID }}"
-                  expires: 30d
-                  channelId: "${{ secrets.DEV_PREVIEW_CHANNEL }}"
+  build_and_preview:
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 10
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Build
+        run: pnpm build --mode="preview"
+        env:
+            VITE_FIREBASE_API_KEY: ${{ secrets.FIREBASE_API_KEY }}
+            VITE_FIREBASE_AUTH_DOMAIN: ${{ secrets.FIREBASE_AUTH_DOMAIN }}
+            VITE_FIREBASE_PROJECT_ID: ${{ secrets.FIREBASE_PROJECT_ID }}
+            VITE_FIREBASE_STORAGE_BUCKET: ${{ secrets.FIREBASE_STORAGE_BUCKET }}
+            VITE_FIREBASE_MESSAGING_SENDER_ID: ${{ secrets.FIREBASE_MESSAGIN_SENDER_ID }}
+            VITE_FIREBASE_APP_ID: ${{ secrets.FIREBASE_APP_ID }}
+            VITE_FIREBASE_MEASUREMENT_ID: ${{ secrets.FIREBASE_MEASUREMENT_ID }}
+            VITE_GOOGLE_CLIENT_ID: ${{ secrets.GOOGLE_CLIENT_ID }}
+            VITE_APP_ENV: preview
+            VITE_APP_CHECK_KEY: ${{ secrets.FIREBASE_APP_CHECK_KEY }}
+            VITE_APPLICATIONS_COLLECTION: ${{ secrets.APPLICATIONS_COLLECTION }}
+            VITE_TICKETS_COLLECTION: ${{ secrets.TICKETS_COLLECTION }}
+            VITE_RSVP_COLLECTION: ${{ secrets.RSVPS_COLLECTION }}
+      - uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: ${{ secrets.GITHUB_TOKEN }}
+          firebaseServiceAccount: ${{ secrets.FIREBASE_SERVICE_ACCOUNT_HAWKHACKS_DASHBOARD }}
+          projectId: hawkhacks-dashboard
+          expires: 30d
```

Comment:
From the ticket acceptance criteria:

> Clean up preview when PR closes.

I couldn't find any option for this. A 30d expiry should work